tsdb/head.go

```diff
@@ -837,6 +837,7 @@ func (h *Head) Truncate(mint int64) (err error) {
 	h.metrics.checkpointCreationTotal.Inc()
 	if _, err = wal.Checkpoint(h.wal, first, last, keep, mint); err != nil {
 		h.metrics.checkpointCreationFail.Inc()
+		h.metrics.walCorruptionsTotal.Inc()
```

Comment:
Checkpointing could fail for various reasons, we should increment only when it's a WAL corruption. There is `wal.CorruptionErr` which you could check for (don't forget `errors.Cause`).